learning_resources/views.py

```diff
@@ -122,6 +122,15 @@ def _get_base_queryset(self, resource_type: str = None) -> QuerySet:
             *([item.value for item in LearningResourceType]),
         )
         lr_query = lr_query.prefetch_related(*prefetches).distinct()
+
+        published_filter = Q(published=True)
+
+        if is_learning_path_editor(self.request) or is_admin_user(self.request):
```

Comment:
I originally thought of this read-only viewset as intended just for browsing published resources, and if one needed to do any CRUD work, a different endpoint would be used (only LearningPath fits that need right now).  I'm fine with discarding that assumption, but one potential issue with this change is that editors/admins will always see both published and unpublished results here for learning paths, and there may be times when it is only appropriate to show published paths (for instance, a carousel of top 10 newest learning paths or top 10 newest resources in general).  One way to avoid this is to add `published` to the `filterset_fields` above, and filter by `?published=true` whenever necessary.  Can you add that in?

Reply:
PS One potential drawback to this though is that using `published=False` would allow any users to see unpublished courses, programs, videos, paths, etc - which may or may not be desirable?  So might need to somehow force `published=True` for anyone except admins (and editors for learningpaths only)

Reply:
Maybe add an `api/v1/learning_resources/published/` endpoint to be used by carousels and such?